mlir/test/Dialect/Arith/canonicalize.mlir

```diff
@@ -714,6 +714,25 @@ func.func @extFPVectorConstant() -> vector<2xf128> {
   return %0 : vector<2xf128>
 }
 
+// CHECK-LABEL: @truncExtf
+//       CHECK-NOT:  truncf
+//       CHECK:   return  %arg0
+func.func @truncExtf(%arg0: f32) -> f32 {
+  %extf = arith.extf %arg0 : f32 to f64
+  %trunc = arith.truncf %extf : f64 to f32
+  return %trunc : f32
+}
+
+// CHECK-LABEL: @truncExtf2
+//       CHECK:  %[[ARG0:.+]]: f32
+//       CHECK:  %[[CST:.*]] = arith.truncf %[[ARG0:.+]] : f32 to f16
```

Comment:
I'm wondering if this is producing the exact same bit pattern. If not, maybe this folding is safe only under fast-math assumptions.

Reply:
I think these are safe in this direction, but I always end-up surprised with floating-point so I would bet too much on it either :)

Reply:
Sadly, I can't alive2 this to be really safe, but it *feels* fine now that I've thought about it